raft/server.cc

```diff
@@ -1242,7 +1275,9 @@ future<> server_impl::read_barrier(seastar::abort_source* as) {
                 logger.trace("[{}] read_barrier leader not ready", _id);
                 co_await wait_for_apply(++applied, as);
             } else if (std::holds_alternative<raft::not_a_leader>(res)) {
-                leader = std::get<not_a_leader>(res).leader;
+                const auto e = std::get<raft::not_a_leader>(res);
+                logger.trace("[{}] got {}, will retry on the next tick", _id, e);
```

Comment:
the message says we will retry on the next tick, but there is no `wait_for_next_tick` call here.